data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetProposerDuties.java

```diff
@@ -13,59 +13,93 @@
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.HTTP_ERROR_RESPONSE_TYPE;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.INTEGER_TYPE;
+import static tech.pegasys.teku.infrastructure.json.types.CoreTypes.UINT64_TYPE;
+import static tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition.listOf;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
-import com.google.common.base.Throwables;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
-import java.util.Map;
 import java.util.Optional;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.jetbrains.annotations.NotNull;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.SyncDataProvider;
 import tech.pegasys.teku.api.ValidatorDataProvider;
 import tech.pegasys.teku.api.response.v1.validator.GetProposerDutiesResponse;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.validator.api.ProposerDuties;
+import tech.pegasys.teku.validator.api.ProposerDuty;
 
-public class GetProposerDuties extends AbstractHandler implements Handler {
-  private static final Logger LOG = LogManager.getLogger();
+public class GetProposerDuties extends MigratingEndpointAdapter {
   private static final String OAPI_ROUTE = "/eth/v1/validator/duties/proposer/:epoch";
   public static final String ROUTE = routeWithBracedParameters(OAPI_ROUTE);
+
+  private static final SerializableTypeDefinition<ProposerDuty> PROPOSER_DUTY_TYPE =
+      SerializableTypeDefinition.object(ProposerDuty.class)
+          .withField("pubkey", PUBLIC_KEY_TYPE, ProposerDuty::getPublicKey)
+          .withField("validator_index", INTEGER_TYPE, ProposerDuty::getValidatorIndex)
+          .withField("slot", UINT64_TYPE, ProposerDuty::getSlot)
+          .build();
+
+  private static final SerializableTypeDefinition<ProposerDuties> RESPONSE_TYPE =
+      SerializableTypeDefinition.object(ProposerDuties.class)
+          .name("GetProposerDutiesResponse")
+          //              .withField("execution_optimistic", BOOLEAN_TYPE, ??) TODO fix?
```

Comment:
yep if its in the standard we should fix here

Reply:
Yes just trying to work out how to get that information, its proving harder than expected

Reply:
would just need to add it to the swagger annotations stuff also so that its in the documentation for now...